google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -135,9 +136,25 @@ public int read(ByteBuffer dst) throws IOException {
     }
   }
 
+  private static boolean isReopenable(Throwable exs) {
+    return exs != null && (exs.getMessage().contains("Connection closed prematurely")
+            || exs.getCause() instanceof javax.net.ssl.SSLException
+            || exs.getCause() instanceof java.io.EOFException
+            || exs.getCause() instanceof java.net.SocketException);
```

Comment:
I think we need both `java.net.SocketException` and `java.net.SocketTimeoutException`, since the latter is not a subclass of the former.